### ISSUE TITLE
Upload does not work when file is large (150mb)

### ISSUE THREAD

**Issue by nstephenh** [janeczku/calibre-web#452]
I've tried to upload a very large (150mb) book via calibre-web and the server times out on the post request before the file finishes uploading. Uploading does work when the filesize is more reasonable (~10 mb).

**Comment by OzzieIsaacs** [comment]
Are you using tornado or gevent? Tornado is failing instantly if the file is larger than 100MB. You can change this in cps.py by changing line 32 from:
http_server = HTTPServer(WSGIContainer(web.app))
to
http_server = HTTPServer(WSGIContainer(web.app),max_buffer_size=1048576000)
(rises limit to 1GB)

**Comment by lucas-dclrcq** [comment]
Same if you use Nginx as a reverse proxy, you need to increase the upload limit and the timeout limit.

e.g add this to your configuration :
```
client_body_in_file_only clean;
client_body_buffer_size 32K;

client_max_body_size 300M;

sendfile on;
send_timeout 300s;
```

**Comment by OzzieIsaacs** [comment]
I increased the upload limit for the "tornado-server" to 200MB, this should be enough according to my experience with books and comics. I didn't find a limit for the gevent server, so I don't know the limit there.

**Comment by wgibcus** [comment]
This issue has raised its head again. The above fix does not work as line 32 does not exist in the current cps.py. Can the upload limit please be something closer to 2Gb? Many comic book compilations have very high file sizes.

**Comment by klausmcm** [comment]
> This issue has raised its head again. The above fix does not work as line 32 does not exist in the current cps.py. Can the upload limit please be something closer to 2Gb? Many comic book compilations have very high file sizes.
> 
> Edit: The location of the buffer size is now located in cps/server.py on line 156. Change the existing value to 2097152000 for a 2Gig limit.

Could this be something that can be configured in an external config file?

**Comment by lhcamilo** [comment]
Also ran into this. I kickstarted the Expanse comic series and I have been storing them in my calibre(web) server until now. 
One of the issues is 202MB, thus the upload fails.

**Comment by OzzieIsaacs** [comment]
Please install gevent (from optional-requirments.txt) and the filesize is unlimited

**Comment by lhcamilo** [comment]
Indeed, that sorts it out.

**Comment by Aquariu** [comment]
Thanks for raising this issue.
In a docker environment, it's unclear for me how I should proceed to correct this issue. Especialy since it looks like inside the docker environement, gevent is already installed:

`root@5ff193747fe3:/# pip3 install gevent
Requirement already satisfied: gevent in /usr/local/lib/python3.10/dist-packages (22.10.2)
Requirement already satisfied: zope.event in /usr/local/lib/python3.10/dist-packages (from gevent) (5.0)
Requirement already satisfied: zope.interface in /usr/local/lib/python3.10/dist-packages (from gevent) (6.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from gevent) (68.0.0)
Requirement already satisfied: greenlet>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from gevent) (2.0.2)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
`
(yes, running as root inside the container was probably a bad idea)

**Comment by lhcamilo** [comment]
all I did was install the latest gevent, from I pip freeze I get 

`gevent==23.7.0`

I also did the update with pip

`pip install -U gevent`

**Comment by Aquariu** [comment]
> Thanks for raising this issue. In a docker environment, it's unclear for me how I should proceed to correct this issue. Especialy since it looks like inside the docker environment, gevent is already installed:
I have successfully solved this on my side and it cleary was not a problem on calibre-web side neither on the docker side but rather on my side.
I have added ,`proxy_max_temp_file_size           1024M;` in the "location" block to my nginx file as well as `client_max_body_size 1024M;`  in the server block. I am unsure as to which one is making this work but I'm happy to report that the problem was indeed on my side as you have suggested repeatedly already. 
Kind regards,
